db/bson_stream.go

```diff
@@ -144,3 +146,7 @@ func (bs *BSONSource) LoadNext() []byte {
 func (bs *BSONSource) Err() error {
 	return bs.err
 }
+
+func (bs *BSONSource) SetMaxBSONSize(size int32) {
+	bs.MaxBSONSize = size
```

Comment:
I think if you're going to do this, you have to also resize the `reusableBuf` if it exists.

I would mildly prefer to have size be set at construction time only, so this sort of resizing isn't necessary, but I don' think that preference has to hold up merging.